src/client/datascience/jupyter/kernels/kernelDependencyService.ts

```diff
@@ -37,12 +40,44 @@ export class KernelDependencyService implements IKernelDependencyService {
         interpreter: PythonEnvironment,
         token?: CancellationToken,
         disableUI?: boolean
-    ): Promise<KernelInterpreterDependencyResponse> {
+    ): Promise<void> {
         traceInfo(`installMissingDependencies ${interpreter.path}`);
         if (await this.areDependenciesInstalled(interpreter, token)) {
-            return KernelInterpreterDependencyResponse.ok;
+            return;
+        }
+
+        // Cache the install run
+        let deferred = this.installPromises.get(interpreter.path);
+        if (!deferred) {
+            deferred = createDeferredFromPromise(this.runInstaller(interpreter, token, disableUI));
+            this.installPromises.set(interpreter.path, deferred);
```

Comment:
```suggestion
        // Cache the install run
        let promise = this.installPromises.get(interpreter.path);
        if (!promise) {
            promise = this.runInstaller(interpreter, token, disableUI);
            this.installPromises.set(interpreter.path, promise);
```

Reply:
We can do away with the deferred and jsut cache the promise.

Reply:
Yeah good point.